test/core/primitives/primitives_codec_test.cpp

```diff
@@ -101,82 +41,95 @@ class Primitives : public testing::Test {
     codec_ = std::make_shared<ScaleCodecImpl>();
   }
 
-  const BlockHeader &header() const {
-    return header_;
-  }
-
-  const Buffer &headerMatch() const {
-    return header_match_;
-  }
-
-  const Extrinsic &extrinsic() const {
-    return extrinsic_;
-  }
-
-  const Buffer &extrinsicMatch() const {
-    return extrinsic_match_;
-  }
-
-  const Block &block() const {
-    return block_;
-  }
-
-  const Buffer &blockMatch() const {
-    return block_match_;
-  }
-
-  const Version &version() const {
-    return version_;
-  }
-
-  const Buffer &versionMatch() const {
-    return version_match_;
-  }
-
-  const BlockId &blockIdNumber() const {
-    return block_id_number_;
-  }
-
-  const BlockId &blockIdHash() const {
-    return block_id_hash_;
-  }
-
-  const Buffer &blockIdNumberMatch() const {
-    return block_id_number_match_;
-  }
-
-  const Buffer &blockIdHashMatch() const {
-    return block_id_hash_match_;
-  }
-
-  const Valid &validTransaction() const {
-    return valid_;
-  }
-
-  const Buffer &validMatch() const {
-    return valid_match_;
-  }
-
-  auto codec() const {
-    return codec_;
-  }
-
- private:
-  BlockHeader header_;
-  Extrinsic extrinsic_;
-  Buffer header_match_;
-  Buffer extrinsic_match_;
-  Block block_;
-  Buffer block_match_;
-  Version version_;
-  Buffer version_match_;
-  BlockId block_id_number_;
+ protected:
+  /// block header
+  BlockHeader block_header_{{1},   // buffer: parent_hash
```

Comment:
Still not clear from comments that block header and encoded header correspond to each other. Better to write like
```
/// Block header and corresponding scale buffer representation
BlockHeader block_header_{...} ;
Buffer enconded_header_{...};
/// Extrinsic and corresponding scale buffer representation
Extrinsic extrinsic_{...};
Buffer encoded_extrinsic_{...};
```

Reply:
Ok